loader/CMakeLists.txt

```diff
@@ -130,14 +131,10 @@ if(WIN32)
             endif()
         endif()
     endif()
-    if (NOT MSVC)
-        set(USE_MASM OFF)
-    endif ()
-    option(USE_MASM "Use MASM" ON)
     if (USE_MASM)
-      enable_language(ASM_MASM)
-    endif ()
-    if(CMAKE_ASM_MASM_COMPILER_WORKS OR JWASM_FOUND)
+        enable_language(ASM_MASM OPTIONAL)
```

Comment:
Why is OPTIONAL being specified?

> The OPTIONAL keyword is a placeholder for future implementation and does not currently work. Instead you can use the [CheckLanguage](https://cmake.org/cmake/help/latest/module/CheckLanguage.html#module:CheckLanguage) module to verify support before enabling.

Reply:
Since I don't have a machine that has visual studio installed, I don't know if it's possible to have the compiler without ml.
The hope here was that should enable_language fail, the `CMAKE_ASM_MASM_COMPILER_WORKS` check below would automatically use the assembly fallback path instead.

From what I could tell, CMake does have some code with [`OPTIONAL`](https://gitlab.kitware.com/cmake/cmake/-/blob/v3.27.6/Source/cmGlobalGenerator.cxx?ref_type=tags#L872), but it is not documented from what I can tell. Additionally, I don't think it works in general as when I test it on my Windows machine that does not have visual studio installed, it still says
```cmake
-- The ASM_MASM compiler identification is unknown
-- Found assembler: ml
-- Configuring incomplete, errors occurred!
```
with and without OPTIONAL. So I think we could remove it.

Reply:
Yes removing it for now seems like the best course of action.

Reply:
```suggestion
        enable_language(ASM_MASM)
```